fri-verifier-air/src/air.rs

```diff
@@ -0,0 +1,181 @@
+use core::borrow::Borrow;
+use core::marker::PhantomData;
+
+use p3_air::{Air, AirBuilder, BaseAir};
+use p3_field::{Field, PrimeCharacteristicRing};
+use p3_matrix::Matrix;
+
+use crate::columns::{num_commit_phase_cols_default, CommitPhaseCols};
+
+/// AIR for FRI commit phase verification.
+///
+/// Verifies one arity-2 FRI folding step per row:
+///   folded_eval = eval₀
+///                + ((β - x₀) * (eval₁ - eval₀)) * (x₁ - x₀)⁻¹
+///                + is_roll_in * (β² * reduced_opening)
+///
+/// Notes:
+/// - Index relations are enforced with an explicit `lsb` bit:
+///     domain_index = 2 * parent_index + lsb
+///     sibling_index = domain_index + (1 - 2*lsb)
+/// - `eval_1` is the sibling evaluation proven against `fri_commit` via an MMCS chip (CTL).
+/// - All extension-field ops over 4-limb representations are expected to be enforced
+///   via CTLs to extension arithmetic chips (add/sub/mul/scale).
+#[derive(Debug)]
+pub struct CommitPhaseAir<F> {
+    _phantom: PhantomData<F>,
+}
+
+impl<F> CommitPhaseAir<F> {
+    pub const fn new() -> Self {
+        Self {
+            _phantom: PhantomData,
+        }
+    }
+}
+
+impl<F> Default for CommitPhaseAir<F> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<F: Field> BaseAir<F> for CommitPhaseAir<F> {
+    fn width(&self) -> usize {
+        num_commit_phase_cols_default()
+    }
+}
+
+impl<AB: AirBuilder> Air<AB> for CommitPhaseAir<AB::F>
+where
+    AB::F: Field,
+{
+    #[inline]
+    fn eval(&self, builder: &mut AB) {
+        let main = builder.main();
+        let local = main
+            .row_slice(0)
+            .expect("main trace matrix should not be empty");
+        let local = (*local).borrow();
+
+        eval_commit_phase_constraints(builder, local);
+        // (Optional) extra sanity checks during development:
+        // eval_debug_constraints(builder, local);
+    }
+}
+
+/// FRI folding constraint evaluation for one row.
+fn eval_commit_phase_constraints<AB: AirBuilder>(
+    builder: &mut AB,
+    local: &CommitPhaseCols<AB::Var, 8>,
+) where
+    AB::Expr: PrimeCharacteristicRing,
+{
+    // =========================
+    // INDEX MANAGEMENT
+    // =========================
+    // lsb ∈ {0,1}
+    builder.assert_eq(local.lsb.clone() * (local.lsb.clone() - AB::Expr::ONE), AB::Expr::ZERO);
```

Comment:
there is an `assert_bool` function
```suggestion
    builder.assert_bool(local.lsb.clone());
```